src/main.cpp

```diff
@@ -93,20 +98,29 @@ int main()
                     updateGameScore(&gameScore, food);
                     snake.addSegment(food.getPosition());
 
-                    std::random_device seeder;
-                    std::mt19937 rng(seeder());
-                    std::uniform_int_distribution<> xDistr(0u, (GAME_FIELD_WIDTH / BLOCK_SIZE) - 1u);
-                    std::uniform_int_distribution<> yDistr(0u, (GAME_FIELD_HEIGHT / BLOCK_SIZE) - 1u);
-                    float xCoord{xDistr(rng) * static_cast<float>(BLOCK_SIZE)};
-                    float yCoord{yDistr(rng) * static_cast<float>(BLOCK_SIZE)};
-                    food.setPosition(sf::Vector2f{xCoord, yCoord});
+                    sf::Vector2f newFoodPosition = generatePosition(gameFieldBounds) * static_cast<float>(BLOCK_SIZE);
+                    food.setPosition(newFoodPosition);
                     while (isIntersectVector2VectorArray(food, getSegments(snake)))
                     {                                 
-                        xCoord = xDistr(rng) * static_cast<float>(BLOCK_SIZE);
-                        yCoord = yDistr(rng) * static_cast<float>(BLOCK_SIZE);
-                        food.setPosition(sf::Vector2f{xCoord, yCoord});
+                        newFoodPosition = generatePosition(gameFieldBounds) * static_cast<float>(BLOCK_SIZE);
+                        food.setPosition(newFoodPosition);
                     }
                     std::cout << "Food Position: " << food.getPosition() << std::endl;
+                    
+                    // TODO: Make type choosing more random or more deterministic
+                    int pseudoRandomNumber = static_cast<int>(food.getPosition().x) * timer.getElapsedTime().asMilliseconds();
+                    if ((pseudoRandomNumber % 3) == 0)
+                    {
+                        food.setType(FoodType::Valuable);
+                    }
+                    else if ((pseudoRandomNumber % 2) == 0)
+                    {
+                        food.setType(FoodType::Aggressive);
+                    }
+                    else 
+                    {
+                        food.setType(FoodType::Neutral);
+                    }
```

Comment:
Why don't you use the same generator that you used before. The frame time in integer milliseconds is not very random, it usually equals to `1000/60`

Reply:
There will be special logic to choose food type